backend/convert_from_protobuf.go

```diff
@@ -70,14 +71,25 @@ func (f ConvertFromProtobuf) DataSourceInstanceSettings(proto *pluginv2.DataSour
 	}
 }
 
+// UserAgent converts protobuf version of a UserAgent to the SDK version.
+func (f ConvertFromProtobuf) UserAgent(u string) *useragent.UserAgent {
+	ua, err := useragent.NewFromString(u)
+	if err != nil {
+		return &useragent.UserAgent{}
```

Comment:
Yeah I think so - at the end of the day this just meta information, and an invalid/empty string should just give empty information rather than error out. I'll made sure to document this though